ci/jenkins/stop-jobs.sh

```diff
@@ -0,0 +1,156 @@
+    # so here "*_anime" means job is in running state.
+    # red_anime: last failed job is in running state.
+    # blue_anime: last successful job is in running state.
+    # aborted_anime: last aborted job is in running state.   
+    job_color=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10][$i].color" | tr -d '"')
+
+    if [[ ${job_color: -5} == "anime" ]]; then
+        last_build_number=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/api/json" | jq "[.[]][10][0].number" | tr -d '"')
+        pr_id=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].ghprbPullId"| tr -d '"')
+
+        if [[ "$PULL_REQUEST_ID" == "$pr_id" ]]; then
+            echo "====== abort running job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+
+            # need to stop first then clean.
+            curl -XPOST "${JENKINS_URL}job/$capv_job_name/$last_build_number/stop" -u admin:$JENKINS_API_TOKEN
+
+            CLUSTER_NAME=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].BUILD_TAG"| tr -d '"')
+            ip_of_capv_testbed=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].SSH_CONNECTION" | awk '{print $3}')
+            
+            cleanup_running_job "$CLUSTER_NAME" "$ip_of_capv_testbed"
+        fi
+    fi   
+}
+
+function stop_kill() {
+    echo "====== checking for stale jobs on PR ${PULL_REQUEST_ID} ======"
+    length_of_capv_jobs=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10]" | jq length)
+
+    for ((i=0; i<$length_of_capv_jobs; i++)); do
+        capv_job_name=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10][$i].name" | tr -d '"')
+
+        # check all stale waiting jobs on PR.
+        abort_queue_jobs "$capv_job_name"
+
+        # check all stale running jobs on PR.
+        abort_running_jobs "$capv_job_name" "$i"
+    done
+}
+
+# ensures that the script can be run from anywhere.
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+GIT_CHECKOUT_DIR=${THIS_DIR}/../..
+pushd "$THIS_DIR" > /dev/null
+
+function release_static_ip {
+    CLUSTER_NAME=$1
+    ip_of_capv_testbed=$2
+    echo '=== Releasing IP ==='
+    ssh -o StrictHostKeyChecking=no -i /var/lib/jenkins/.ssh/id_rsa -n jenkins@${ip_of_capv_testbed} "cat "$DEFAULT_WORKDIR/host-local.json" | \
+    CNI_COMMAND=DEL CNI_CONTAINERID="$CLUSTER_NAME" \
+    CNI_NETNS=/dev/null CNI_IFNAME=dummy0 CNI_PATH=. /usr/bin/host-local"
+    echo Released IP
+}
+
+function cleanup_running_job {
+    CLUSTER_NAME=$1
+    ip_of_capv_testbed=$2
+    echo "=== Cleaning up VMC cluster ${CLUSTER_NAME} ==="
+    # To run kubectl cmds.
+    ssh -o StrictHostKeyChecking=no -i /var/lib/jenkins/.ssh/id_rsa -n jenkins@${ip_of_capv_testbed} kubectl delete ns ${CLUSTER_NAME}
```

Comment:
It should delete all cluster CRs in the namespace first. Deleting the namespace directly may cause some dangling resources, and leads to deletion failure.